src/ALCProxy.Communication/ICommObjects.cs

```diff
@@ -0,0 +1,257 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. 
+// The .NET Foundation licenses this file to you under the MIT license. 
+// See the LICENSE file in the project root for more information. 
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Loader;
+
+namespace ALCProxy.Communication
+{
+    /// <summary>
+    /// This currently is designed to only work in-process
+    /// TODO: set up to allow for construction of out-of-proc proxies
+    /// </summary>
+    public abstract class ALCClient : IProxyClient
+    {
+        //Can't make this an IServerObject directly due to the type-loading barrier
+        protected object _server;
+        protected string _serverTypeName;
+        protected Type _intType;
+        protected MethodInfo _callMethod;
+        public ALCClient(Type interfaceType, string serverName)
+        {
+            _intType = interfaceType;
+            _serverTypeName = serverName;
+        }
+        private Type FindTypeInAssembly(string typeName, Assembly a)
+        {
+            //find the type we're looking for
+            Type t = null;
+            foreach (Type ty in a.GetTypes())
+            {
+                if (ty.Name.Equals(typeName) || (ty.Name.StartsWith(typeName) && ty.Name.Contains("`"))) // will happen for non-generic types, generics we need to find the additional "`1" afterwards
```

Comment:
>generics we need to find the additional "`1" afterwards

What about generic types with more than 1 parameter? I would just clarify.

Reply:
I've removed this part when getting the method to work with `Assembly.GetType`